### PR TITLE
GEOS-7215 : Reimplemented on Geoserver version 2.8.x

### PR DESCRIPTION
**Pull request by anshumannayak1**
Dear Team,
I am re-implemented the issue "Geoserver WFS GetFeature requests return incorrect result" in the geoserver version 2.8.x. I have described all the details in the the issue GEOS-7215.

Warm Regards
Anshuman Nayak

### CODE REVIEW

**Comment by bradh** [comment]
I think there was a requirement on the JIRA ticket for a test (https://osgeo-org.atlassian.net/browse/GEOS-7215?focusedCommentId=49526&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-49526) that verifies the fix. I don't see that in the proposed change.

File: src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java [hunk 1]
```diff
@@ -563,9 +563,10 @@ public FeatureCollectionResponse run(GetFeatureRequest request)
                     isNumberMatchedSkipped = meta.getSkipNumberMatched()
                             && !request.isResultTypeHits();
                 if (!isNumberMatchedSkipped) {
+                		// Fix for GEOS-7233: totalFeatures for WFS 2.0 request returns wrong count.
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java]: Please remove all these comments, we cannot have the source code riddled with "fix for blah blah" comments

File: src/wfs/src/main/java/org/geoserver/wfs/kvp/GetFeatureKvpRequestReader.java [hunk 2]
```diff
@@ -186,6 +186,12 @@ public Object read(Object request, Map kvp, Map rawKvp) throws Exception {
                         typeNames.add(parsed);
                     }
                 }
+                
+                // Fix for GEOS-7215 : incorrect result from WFS request
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/kvp/GetFeatureKvpRequestReader.java]: Remove comments

**Comment by aaime** [comment]
Your changes contain tabs, please check the coding conventions, we only indent with spaces.
Also, github complains the pull request has conflicts

File: src/wfs/src/main/java/org/geoserver/wfs/xml/XSProfile.java [hunk 3]
```diff
@@ -47,6 +47,8 @@
         proper.add(name(XS.INTEGER)); //BigInteger.class
         proper.add(name(XS.DECIMAL)); //BigDecimal.class
         proper.add(name(XS.ANYURI)); //URI.class
+        // Fix for GEOT-5235 : WFS request for format GML3.2 generate exception for UUID
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/xml/XSProfile.java]: This is a different fix, should go in its own pull request, with a test

File: src/wfs/src/main/java/org/geoserver/wfs/kvp/GetFeatureKvpRequestReader.java (hunk 2 above)
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/kvp/GetFeatureKvpRequestReader.java]: Also, one fix per pull request, with its own test

File: src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java (hunk 1 above)
Comment (anshumannayak1) [inline src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java]: I will remove comments and update the request soon.

File: src/wfs/src/main/java/org/geoserver/wfs/kvp/GetFeatureKvpRequestReader.java (hunk 2 above)
Comment (anshumannayak1) [inline src/wfs/src/main/java/org/geoserver/wfs/kvp/GetFeatureKvpRequestReader.java]: Yes you are right, Will do a new pull request for this.

File: src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java (hunk 1 above)
Comment (anshumannayak1) [inline src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java]: comment removed

**Comment by aaime** [comment]
I'm closing this one, it still tries to fit multiple changes in one pull request, and has merge failures. Please make a new set of pull request, with one fix per pull request, and each one having its own tests.
